sale_substate/tests/test_sale_substate.py

```diff
@@ -0,0 +1,52 @@
+# Copyright 2019 Akretion Mourad EL HADJ MIMOUNE
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo.tests.common import TransactionCase
+from odoo.exceptions import ValidationError
+
+
+class TestBaseSubstate(TransactionCase):
+
+    def setUp(self):
+        super(TestBaseSubstate, self).setUp()
+        self.substate_test_sale = self.env['sale.order']
+        self.substate_test_sale_line = self.env['sale.order.line']
+
+        self.substate_under_nego = self.env.ref(
+            'sale_substate.base_substate_under_nego')
+        self.substate_won = self.env.ref('sale_substate.base_substate_won')
+        self.substate_wait_docs = self.env.ref(
+            'sale_substate.base_substate_wait_docs')
+        self.substate_valid_docs = self.env.ref(
+            'sale_substate.base_substate_valid_docs')
+        self.substate_in_delivery = self.env.ref(
+            'sale_substate.base_substate_in_delivery')
+
+    def test_sale_order_substate(self):
+        partner = self.env.ref('base.res_partner_1')
+        so_test1 = self.substate_test_sale.create({
+            'name': 'Test base substate to basic sale',
+            'partner_id': partner.id,
+            'line_ids': [(0, 0, {
```

Comment:
Looks like this test is on the mock up model, `base.substate.test.sale`
In sale.order, there is no `line_ids`, but rather `order_line`, is n't it?